### PR TITLE
feat: Add a TS function to transfer user share to a given Sui public key

### PR DESCRIPTION
**Pull request by yaelAbergel**
addresses #171

### CODE REVIEW

File: sdk/typescript/test/e2e/dwallet-transfer.test.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
+import { verify_user_share } from '@dwallet-network/signature-mpc-wasm';
+import { beforeAll, describe, expect, it } from 'vitest';
+
+import {
+	createActiveEncryptionKeysTable,
+	createDWallet,
+	decrypt_user_share,
+	EncryptionKeyScheme,
+	generate_keypair,
+	generate_proof,
```
Comment (ItayLevyOfficial) [inline sdk/typescript/test/e2e/dwallet-transfer.test.ts:10]: write which test has been moved

**Comment by yaelAbergel** [comment]
note: In this pr I moved two tests from `signature-mpc.test.ts` into `dwallet-transfer.ts` (the tests: 1. "Create public key" 2. "Test key share transfer").

File: sdk/typescript/test/e2e/signature-mpc.test.ts
Comment (yaelAbergel) [inline sdk/typescript/test/e2e/signature-mpc.test.ts:1]: note: In this pr I moved two tests from this file into dwallet-transfer.ts (the tests: 1. "Create public key" 2. "Test key share transfer").

File: sdk/typescript/test/e2e/encrypt-user-share.test.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+			dkg!,
+			toolbox.keypair.getPublicKey(),
+			encryptionKeysHolder.objectId,
+		);
+
+		const decryptedKeyShare = decrypt_user_share(
+			encryptionKey,
+			decryptionKey,
+			encryptedUserShareAndProof,
+		);
```
Comment (ItayLevyOfficial) [inline sdk/typescript/test/e2e/encrypt-user-share.test.ts:159]: You are using the wrong `encryptedUserShareAndProof`; you should use the one that you are storing on chain from `sendUserShareToAddress`, not a new one that you are generating in this function as you do

File: sdk/typescript/src/signature-mpc/encrypt_user_share.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import type { DWalletClient } from '../client/index.js';
+import type { Keypair, PublicKey } from '../cryptography/index.js';
+import {
+	getActiveEncryptionKeyObjID,
+	getEncryptionKeyByObjectId,
+	transferEncryptedUserShare,
+} from './dwallet';
+import { Dwallet, generate_proof } from './dwallet_2pc_mpc_ecdsa_k1_module';
+
+export const sendUserShareToAddress = async (
```
Comment (ItayLevyOfficial) [inline sdk/typescript/src/signature-mpc/encrypt_user_share.ts:12]:
```suggestion
export const sendUserShareToSuiPubKey = async (
```

**Review by ItayLevyOfficial — APPROVED** [review]
LGTM!

File: sdk/typescript/src/signature-mpc/encrypt_user_share.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+		recipientData?.encryptionKey!,
+	);
+	return await transferEncryptedUserShare(
+		client,
+		keypair,
+		encryptedUserShareAndProof,
+		activeEncryptionKeyObjID,
+		dwallet.dwalletId,
+	);
+};
```
Comment (ycscaly) [inline sdk/typescript/src/signature-mpc/encrypt_user_share.ts:46]: Really good job!

**Review by ycscaly — APPROVED** [review]
Flawless!

**Review by zeevmoney — APPROVED** [review]

**Pull request merged**